core/blockchain.go

```diff
@@ -3015,3 +3034,155 @@ func EnablePersistDiff(limit uint64) BlockChainOption {
 		return chain
 	}
 }
+
+func (bc *BlockChain) GetRootByDiffHash(blockNumber uint64, blockHash common.Hash, diffHash common.Hash) (*types.VerifyResult, error) {
```

Comment:
It is not proper to return a `*types.VerifyResult` while the function name is `GetRootByDiffHash`. Try better name?

Reply:
`VerifyResult` is a too common name in `types` package, need better name as well.

Reply:
Actually, I plan move the verify related structure into the verify_manager.go after fast node has completed.